experimental/PropertyDDS/packages/property-proxy/src/arrayProxyHandler.js

```diff
@@ -21,35 +21,35 @@ import { Utilities } from './utilities';
  * @hidden
  */
 function setLength(target, length) {
-  const newLength = Number(length) === length ? parseInt(length, 10) : 0;
-  if (newLength < 0) {
-    throw new RangeError('Invalid array length');
-  }
-
-  const property = target.getProperty();
-  const currentLength = property.getLength();
-
-  if (newLength === 0) {
-    property.clear();
-  } else if (currentLength > newLength) {
-    // Shorten the array
-    Utilities.wrapWithPushPopNotificationDelayScope(property,
-      () => property.removeRange(newLength, currentLength - newLength));
-  } else if (currentLength < newLength) {
-    // Fill the array with empty but valid values (instead of 'undefined')
-    const itemProps = [];
-    for (let i = currentLength; i < newLength; i++) {
-      if (PropertyFactory.instanceOf(property, 'Reference', 'array')) {
-        itemProps.push('');
-      } else {
-        itemProps.push(PropertyFactory.create(property.getTypeid()));
-      }
+    const newLength = Number(length) === length ? parseInt(length, 10) : 0;
+    if (newLength < 0) {
+        throw new RangeError('Invalid array length');
     }
-    Utilities.wrapWithPushPopNotificationDelayScope(property, () => {
```

Comment:
FYI - I'm having some trouble following the type coercion here.

It looks like:
* If you pass a number, we invoke parseInt()
* If you pass a string, you get a 0

Reply:
yeah its definetly odd, I address this in a following PR, at least partially currently I try to keep the behaviour unchanged as it works but there are also other instance where things look dicy.